cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/mongo/impl/PublicationSeachProviderImpl.java

```diff
@@ -0,0 +1,51 @@
+package edu.asu.diging.cord19.explorer.core.mongo.impl;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.cord19.explorer.core.model.impl.PublicationImpl;
+import edu.asu.diging.cord19.explorer.core.mongo.PublicationSearchProvider;
+
+@Service
+public class PublicationSeachProviderImpl implements PublicationSearchProvider {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Override
+    public List<PublicationImpl> searchPublicationTitles(String title) {
+        Criteria regex = Criteria.where("metadata.title").regex(".*" + title + ".*", "i");
+        List<PublicationImpl> results = mongoTemplate.find(new Query().addCriteria(regex), PublicationImpl.class);
+        return results;
+    }
+
+    @Override
+    public Page<PublicationImpl> paginateResults(Pageable pageable, List<PublicationImpl> pubs) {
```

Comment:
You need to get the page of results from the db, not create it yourself. The point of getting pages rather than all results is to avoid performance and memory issues, which won't be avoided the way you implemented it.